pkg/generic/descriptor/http.go

```diff
@@ -32,25 +35,124 @@ const (
 	MIMEApplicationProtobuf = "application/x-protobuf"
 )
 
+var (
+	_ dhttp.RequestGetter  = &HTTPRequest{}
+	_ dhttp.ResponseSetter = &HTTPResponse{}
+)
+
 // HTTPRequest ...
 type HTTPRequest struct {
-	Header      http.Header
-	Query       url.Values
-	Cookies     Cookies
-	Method      string
-	Host        string
-	Path        string
 	Params      *Params // path params
+	Request     *http.Request
 	RawBody     []byte
 	Body        map[string]interface{}
 	GeneralBody interface{} // body of other representation, used with ContentType
 	ContentType MIMEType
+	cookies     map[string]string
+	query       url.Values
```

Comment:
Why cache is needed?

Reply:
In order to make it compatible with the original kitex code; for cookie: [pkg/generic/httpthrift_codec.go](https://github.com/cloudwego/kitex/blob/develop/pkg/generic/httpthrift_codec.go#L140) & [pkg/generic/httppbthrift_codec.go](https://github.com/cloudwego/kitex/blob/develop/pkg/generic/httppbthrift_codec.go#L154), for query: [pkg/generic/httpthrift_codec.go](https://github.com/cloudwego/kitex/blob/develop/pkg/generic/httpthrift_codec.go#LL132C6-L132C6).
(you can take a look at @felix021 's comments)

Reply:
It's for performance, since `URL.Query()` parses the query string each time it's called, so a cached parse result will be faster if `GetQuery(key)` is to be called multiple times for a single request.